signal_timeout_block.py

```diff
@@ -67,8 +68,10 @@ def start(self):
                         # scheduled before this persisted job is scheduled, so
                         # only do this if one has not been scheduled already.
                         self._schedule_timeout_job(job, key, interval, True)
+        self._persistence_scheduled.set()
```

Comment:
Oh yeah, that's a good point. If a signal came in and created the timeout job but then the persisted timeout job overwrote it that would probably be a bad thing. But I think the logic in the `start` method already tries to avoid that situation by checking if the key has been set in the dictionary.